src/Databases/DatabaseAtomic.cpp

```diff
@@ -483,33 +496,41 @@ UUID DatabaseAtomic::tryGetTableUUID(const String & table_name) const
 
 void DatabaseAtomic::beforeLoadingMetadata(ContextMutablePtr /*context*/, LoadingStrictnessLevel mode)
 {
+    auto shared_disk = getContext()->getSharedDisk();
     if (mode < LoadingStrictnessLevel::FORCE_RESTORE)
         return;
 
-    if (!fs::exists(path_to_table_symlinks))
+    if (!shared_disk->isSymlinkSupported())
+        return;
+
+    if (!shared_disk->existsFileOrDirectory(path_to_table_symlinks))
         return;
 
     /// Recreate symlinks to table data dirs in case of force restore, because some of them may be broken
-    for (const auto & table_path : fs::directory_iterator(path_to_table_symlinks))
+    for (const auto dir_it = shared_disk->iterateDirectory(path_to_table_symlinks); dir_it->isValid(); dir_it->next())
     {
-        if (!FS::isSymlink(table_path))
+        auto table_path = fs::path(dir_it->path());
+        if (table_path.filename().empty())
+            table_path = table_path.parent_path();
```

Comment:
why this is now needed?

Reply:
In the following logic, previously, we used `FS::isSymink`. It did the same check.
Now we use `shared_disk->isSymlink()`, it doesn't  check if the path ends with `/`.
And it will return `false` in that case.